web/js/modules/geosearch/selectors.js

```diff
@@ -0,0 +1,200 @@
+  return coordinatesWithinExtent;
+}
+
+// add coordinates marker
+export function addCoordinatesMarker(activeMarker, config, map, coordinates, reverseGeocodeResults) {
+  const { projections } = config;
+  const { selected } = map.ui;
+  const { proj } = selected;
+  const { crs } = projections[proj];
+
+  // remove any markers
+  removeCoordinatesMarker(activeMarker, map);
+
+  // only add marker within current map extent
+  const coordinatesWithinExtent = areCoordinatesWithinExtent(map, config, coordinates);
+  if (!coordinatesWithinExtent) {
+    return false;
+  }
+
+  // transform coordinates if not CRS EPSG:4326
+  let transformedCoordinates = false;
+  if (proj !== 'geographic') {
+    transformedCoordinates = coordinatesCRSTransform(coordinates, 'EPSG:4326', crs);
+  }
+
+  // create Ol vector layer map pin
+  const marker = createPin(coordinates, transformedCoordinates, reverseGeocodeResults);
+  selected.addLayer(marker);
+  selected.renderSync();
+  return marker;
+}
+
+// remove coordinates tooltip from all projections
+export function removeCoordinatesOverlayFromAllProjections(proj) {
+  const mapProjections = Object.keys(proj);
+  mapProjections.forEach((mapProjection) => {
+    const mapOverlays = proj[mapProjection].getOverlays().getArray();
+    const coordinatesTooltipOverlay = mapOverlays.filter((overlay) => {
+      const { id } = overlay;
+      return id && id.includes('coordinates-map-marker');
+    });
+    if (coordinatesTooltipOverlay.length > 0) {
+      proj[mapProjection].removeOverlay(coordinatesTooltipOverlay[0]);
+    }
+  });
+}
+
+// remove coordinates marker
+export function removeCoordinatesMarker(activeMarker, map) {
+  const { selected, proj } = map.ui;
+  // remove marker
+  if (activeMarker) {
+    activeMarker.setMap(null);
+    selected.removeLayer(activeMarker);
+  }
+  // remove tooltip from all projections
+  removeCoordinatesOverlayFromAllProjections(proj);
+}
+
+// create Ol vector layer map pin
+const createPin = function(coordinates, transformedCoordinates = false, reverseGeocodeResults = {}) {
+  const [longitude, latitude] = coordinates;
+  const iconFeature = new OlFeature({
+    geometry: new OlPoint(transformedCoordinates || coordinates),
+    reverseGeocodeResults,
+    latitude,
+    longitude,
+  });
+
+  const iconStyle = new OlStyle({
+    image: new OlIcon({
+      anchorOrigin: 'bottom-left',
+      anchorXUnits: 'fraction',
+      anchorYUnits: 'pixels',
+      scale: 0.5,
+      src: 'images/map-pin.png',
+    }),
+  });
+
+  iconFeature.setStyle(iconStyle);
+  iconFeature.setId('coordinates-map-marker');
+
+  const vectorSource = new OlSourceVector({
+    wrapX: false,
+    features: [iconFeature],
+  });
+  const vectorLayer = new OlLayerVector({
+    source: vectorSource,
+  });
+
+  return vectorLayer;
+};
+
+// ArcGIS World Geocoding Service API Requests and Options
+const GEOCODE_OPTIONS = {
+  urlBase: 'https://geocode.arcgis.com/arcgis/rest/services/World/GeocodeServer/',
```

Comment:
Should this URL go into `features.json` along with our events, notifications, and snapshots API URLs?